packages/google_maps_flutter/google_maps_flutter_web/lib/src/marker_clustering.dart

```diff
@@ -121,15 +126,24 @@ class ClusterManagersController extends GeometryController {
 
   /// Converts [MarkerClustererCluster] to [Cluster].
   Cluster _convertCluster(ClusterManagerId clusterManagerId,
-      MarkerClustererCluster markerClustererCluster) {
+      MarkerClustererCluster<T> markerClustererCluster) {
     final LatLng position = gmLatLngToLatLng(markerClustererCluster.position);
     final LatLngBounds bounds =
-        gmLatLngBoundsTolatLngBounds(markerClustererCluster.bounds!);
+        gmLatLngBoundsToLatLngBounds(markerClustererCluster.bounds!);
+
+    final List<MarkerId> markerIds =
+        markerClustererCluster.markers.map<MarkerId>((T marker) {
+      return getOnMarkerType(
+        marker: marker,
+        legacy: (gmaps.Marker marker) {
+          return MarkerId((marker.get('markerId')! as JSString).toDart);
+        },
+        advanced: (gmaps.AdvancedMarkerElement marker) {
+          return MarkerId((marker.getAttribute('id')! as JSString).toDart);
+        },
+      );
```

Comment:
Having to use a `getOnMarkerType` here is unfortunate, but since this is the only case, you can just inline the `isA` check and return the correct type.

Ideally you could wrap this as a very specific `getMarkerId(JSObject marker)` method in `convert.dart` that encapsulates the exact logic that we need, instead of having a callback-based general method.